service/worker/archiver/util.go

```diff
@@ -122,11 +122,29 @@ func historyMutated(historyBlob *HistoryBlob, request *ArchiveRequest) bool {
 func validateArchivalRequest(request *ArchiveRequest) error {
 	if len(request.BucketName) == 0 {
 		// this should not be able to occur, if domain enables archival bucket should always be set
-		return cadence.NewCustomError(errEmptyBucket)
+		return cadence.NewCustomError(errInvalidRequest, errEmptyBucket)
 	}
 	return nil
 }
 
+func getUploadHistoryActivityResponse(progress uploadProgress, err error) (uploadResult, error) {
```

Comment:
Lets chat about different approaches to this method. This does not really feel like a utility method to me with how tightly these error types are coupled to the core logic of upload method.